internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource.go

```diff
@@ -0,0 +1,700 @@
+			DaysOfMonth:      nil,
+			DaysOfTheWeek:    pointer.To(daysOfWeek),
+			MonthsOfYear:     pointer.To(monthsOfYear),
+			ScheduleTimes:    pointer.To(scheduleTimes),
+			WeeksOfTheMonth:  pointer.To(weeksOfMonth),
+		})
+	}
+
+	return pointer.To(results)
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerBackupRules(input []backuppolicies.BasePolicyRule) []string {
+	backupRules := make([]string, 0)
+
+	for _, item := range input {
+		if v, ok := item.(backuppolicies.AzureBackupRule); ok {
+			if v.Trigger != nil {
+				if scheduleBasedTrigger, ok := v.Trigger.(backuppolicies.ScheduleBasedTriggerContext); ok {
+					backupRules = scheduleBasedTrigger.Schedule.RepeatingTimeIntervals
+					return backupRules
+				}
+			}
+		}
+	}
+
+	return backupRules
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerBackupTimeZone(input []backuppolicies.BasePolicyRule) string {
+	var timeZone string
+
+	for _, item := range input {
+		if backupRule, ok := item.(backuppolicies.AzureBackupRule); ok {
+			if backupRule.Trigger != nil {
+				if scheduleBasedTrigger, ok := backupRule.Trigger.(backuppolicies.ScheduleBasedTriggerContext); ok {
+					timeZone = pointer.From(scheduleBasedTrigger.Schedule.TimeZone)
+					return timeZone
+				}
+			}
+		}
+	}
+
+	return timeZone
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule(input []backuppolicies.BasePolicyRule) []BackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule {
+	results := make([]BackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule, 0)
+
+	for _, item := range input {
+		if retentionRule, ok := item.(backuppolicies.AzureRetentionRule); ok {
+			if pointer.From(retentionRule.IsDefault) {
+				var lifeCycle []BackupPolicyPostgreSQLFlexibleServerLifeCycle
+				if v := retentionRule.Lifecycles; len(v) > 0 {
+					lifeCycle = flattenBackupPolicyPostgreSQLFlexibleServerLifeCycles(v)
+				}
+
+				results = append(results, BackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule{
+					LifeCycle: lifeCycle,
+				})
+			}
+		}
+	}
+
+	return results
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerRetentionRules(input []backuppolicies.BasePolicyRule) []BackupPolicyPostgreSQLFlexibleServerRetentionRule {
+	results := make([]BackupPolicyPostgreSQLFlexibleServerRetentionRule, 0)
+	var taggingCriterias []backuppolicies.TaggingCriteria
+
+	for _, item := range input {
+		if backupRule, ok := item.(backuppolicies.AzureBackupRule); ok {
+			if trigger, ok := backupRule.Trigger.(backuppolicies.ScheduleBasedTriggerContext); ok {
+				if trigger.TaggingCriteria != nil {
+					taggingCriterias = trigger.TaggingCriteria
+				}
+			}
+		}
+	}
+
+	for _, item := range input {
+		if retentionRule, ok := item.(backuppolicies.AzureRetentionRule); ok {
+			var name string
+			var taggingPriority int
+			var taggingCriteria []BackupPolicyPostgreSQLFlexibleServerCriteria
+
+			if retentionRule.IsDefault == nil || !pointer.From(retentionRule.IsDefault) {
```

Comment:
if `retentionRule.IsDefault == nil`, then `pointer.From(retentionRule.IsDefault) == false`, so:
```suggestion
			if !pointer.From(retentionRule.IsDefault)  {
```

Reply:
Updated